query-graphs/src/postgres.ts

```diff
@@ -0,0 +1,353 @@
+        const childKeys = ["Plan", "Plans"];
+        for (const key of childKeys) {
+            if (!node.hasOwnProperty(key)) {
+                continue;
+            }
+            const child = convertPostgres(node[key], key);
+            if (Array.isArray(child)) {
+                explicitChildren = explicitChildren.concat(child);
+            } else {
+                explicitChildren.push(child);
+            }
+        }
+
+        // Add the following keys as children only when they refer to objects and display as properties if not
+        const objectKeys = [""];
+        for (const key of objectKeys) {
+            if (!node.hasOwnProperty(key)) {
+                continue;
+            }
+            if (typeof node[key] !== "object") {
+                properties[key] = forceToString(node[key]);
+                continue;
+            }
+            const child = convertPostgres(node[key], key);
+            if (Array.isArray(child)) {
+                explicitChildren = explicitChildren.concat(child);
+            } else {
+                explicitChildren.push(child);
+            }
+        }
+
+        // Display these properties always as properties, even if they are more complex
+        const propertyKeys = [""];
+        for (const key of propertyKeys) {
+            if (!node.hasOwnProperty(key)) {
+                continue;
+            }
+            properties[key] = forceToString(node[key]);
+        }
+
+        // Display all other properties adaptively: simple expressions are displayed as properties, all others as part of the tree
+        const handledKeys = tagKeys.concat(childKeys, objectKeys, propertyKeys);
+        for (const key of Object.getOwnPropertyNames(node)) {
+            if (handledKeys.indexOf(key) !== -1) {
+                continue;
+            }
+
+            // Try to display as string property
+            const str = tryToString(node[key]);
+            if (str !== undefined) {
+                properties[key] = str;
+                continue;
+            }
+
+            // Display as part of the tree
+            const innerNodes = convertPostgres(node[key], key);
+            if (Array.isArray(innerNodes)) {
+                additionalChildren.push({tag: key, children: innerNodes});
+            } else {
+                additionalChildren.push({tag: key, children: [innerNodes]});
+            }
+        }
+
+        // Display the cardinality on the links between the nodes
+        let edgeLabel: string | undefined = undefined;
+        if (node.hasOwnProperty("Plan Rows") && typeof node["Plan Rows"] === "number") {
+            edgeLabel = formatMetric(node["Plan Rows"]);
+            if (node.hasOwnProperty("Actual Rows") && typeof node["Actual Rows"] === "number") {
+                //const num0 = Number(node["Plan Rows"]);
+                //const num1 = Number(node["Actual Rows"]);
+                //if (num0 > num1 * 10 || num0 * 10 < num1) {
```

Comment:
why prefer the "computeOrder" over this commented out condition?
Afaict, the `num0 > num1 * 10 || num0 * 10 < num1` would better correspond to the behavior expected by the user.

E.g. Plan Rows = 99 and Actual Rows = 100, would be highlighted by `order0 !== order1`, although being an estimate which is off-by-one, is perfectly fine...
`num0 > num1 * 10 || num0 * 10 < num1` would not highlight this case (as I would expect)

Reply:
agreed